### PR TITLE
Remodeling of range for analyte_cateogry

### LINKED ISSUE #2154: `berkeley` breakdown AnalyteCategoryEnum into types allowable for NucleotideSequencing and MassSpectrometry
Not required before the rollout but at some point we should break down AnalyteCategoryEnum into enumerations specific to subcategories. At present it would be valid to have  analyte_category populated with nom for class NucleotideSequencing

cc @mslarae13

### PR DESCRIPTION
**Pull request by samobermiller**
This PR replaces AnalyteCategoryEnum with NucelotideSequencingEnum and MassSpectrometryEnum and adds titles to the corresponding permissible values
fixes #2154

### CODE REVIEW

**Comment by kheal** [comment]
@aclum and @turbomam FYI, neither @samobermiller  or I could actually get the inheritance on the "sub Enums" to result in properly validating or invalidating examples (see this example src/data/invalid/MassSpectrometry-working.yaml in this branch validates but it should not). 

We tried a few different things to no avail. We only opened this PR to see if the documentation would render appropriately (it doesn't).  Just to give you a heads up about this branch and PR.  I don't think @samobermiller  or I plan to pick it back up anytime soon (we were using it as a training example for how to write and test example yaml data).

**Comment by aclum** [comment]
This does not appear to be working properly based on the passing invalid data MassSpectrometry-working.yaml, it should fail b/c of an invalid analyte_category PV.

**Comment by aclum** [comment]
@kheal  yes, I see that now. @sierra-moxon  or @turbomam  can you take over here?

**Comment by aclum** [comment]
cc @pkalita-lbl  in case you have any linkml insights

**Comment by pkalita-lbl** [comment]
The JSON Schema generator only handles explicitly defined permissible values at the moment: https://github.com/linkml/linkml/blob/5cc5a16e345d3ee5233a193eba4909a8999b43db/linkml/generators/jsonschemagen.py#L426-L457

So, in particular, it doesn't have any logic yet to deal with `inherits` or `minus` as used in this changes.

**Comment by aclum** [comment]
I will wait until the 12/16 release to merge this in  but @turbomam  @sierra-moxon @kheal @mslarae13  I need at least one reviewer on this

File: src/schema/basic_slots.yaml
Comment (sierra-moxon) [inline src/schema/basic_slots.yaml:416]: 👍 - so for now, we fall back to "string" for the range of `analyte_category` if not otherwise defined in `slot_usage`.

**Review by sierra-moxon — APPROVED** [review]

**Review by kheal — APPROVED** [review]

**Pull request merged**
